### PR TITLE
Refs Task#376104 [User] Unit test list bots

### PR DESCRIPTION
**Pull request by nguyenthinhhn**
## Related Tickets

- [#376104](https://dev.sun-asterisk.com/issues/376104)

## What's this PR do ?

- [x] Test list bot

## Performance

- [ ] Resolved n + 1 query
- [ ] Run explain query already
- [ ] Time run rake task : 1000 ms

## Checklist

- [ ] It was tested in local success?
- [ ] Fill link PR into ticket and the opposite
- [ ] Note purpose/reason, solution, scope of influence into ticket

## ENV note

```
```

## Notes
*(Other notes)*

### CODE REVIEW

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.13</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.11</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.9</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.15</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.7</td><td>success</td><td></td></tr></table>

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>php_artisan_migrate_._start_run_.15</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.9</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.13</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.7</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.11</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr></table>

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr></table>

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr></table>

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>failure</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr></table>

File: app/Repositories/Eloquents/BotRepository.php [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    {
+        return Bot::class;
+    }
+
+    public function getAllByUser()
+    {
+        return Auth::user()->bots()
+                            ->orderBy('bots.created_at', 'desc')
+                            ->get();
+    }
```
Comment (framgiaciv3) [inline app/Repositories/Eloquents/BotRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Classes.ClassDeclaration.CloseBraceAfterBody</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/BotRepository.php</br>**Line**: 23</br>The closing brace for the class must go on the next line after the body

File: app/Providers/AppServiceProvider.php [hunk 2]
```diff
@@ -10,7 +10,8 @@
 class AppServiceProvider extends ServiceProvider
 {
     protected $repositories = [
-        'WebhookRepository'
+        'WebhookRepository',
+        'BotRepository'
```
Comment (framgiaciv3) [inline app/Providers/AppServiceProvider.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.NoCommaAfterLast</br>**Severity**: error</br>**File**: app/Providers/AppServiceProvider.php</br>**Line**: 14</br>Comma required after last value in array declaration

File: app/Http/Controllers/BotController.php [hunk 3]
```diff
@@ -5,14 +5,21 @@
 use Auth;
 use App\Models\Bot;
 use Illuminate\Http\Request;
+use App\Repositories\Interfaces\BotRepositoryInterface as BotRepository;
 
 class BotController extends Controller
 {
+	private $botRepository;
```
Comment (framgiaciv3) [inline app/Http/Controllers/BotController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.WhiteSpace.DisallowTabIndent.TabsUsed</br>**Severity**: error</br>**File**: app/Http/Controllers/BotController.php</br>**Line**: 12</br>Spaces must be used to indent lines; tabs are not allowed

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>failure</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr></table>

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>failure</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr></table>

File: app/Http/Controllers/BotController.php (hunk 3 above)
Comment (framgiaciv3) [inline app/Http/Controllers/BotController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.WhiteSpace.DisallowTabIndent.TabsUsed</br>**Severity**: error</br>**File**: app/Http/Controllers/BotController.php</br>**Line**: 12</br>Spaces must be used to indent lines; tabs are not allowed

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr></table>

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr></table>

File: app/Repositories/Interfaces/BotRepositoryInterface.php [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+namespace App\Repositories\Interfaces;
+
+interface BotRepositoryInterface
+{
+    /**
+     * Get all bot by user
+     * @return mixed
+     */
+
```
Comment (quanghung97) [inline app/Repositories/Interfaces/BotRepositoryInterface.php]: xóa 1 dòng

**Review by quanghung97 — APPROVED** [review]
OK

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr></table>

File: app/Http/Controllers/BotController.php [hunk 5]
```diff
@@ -5,14 +5,21 @@
 use Auth;
 use App\Models\Bot;
 use Illuminate\Http\Request;
+use App\Repositories\Interfaces\BotRepositoryInterface as BotRepository;
```
Comment (ghost) [inline app/Http/Controllers/BotController.php:8]: cho a hỏi ngu phát, vì sao mình lại dùng interface mà ko dùng class BotRepository vậy e

File: tests/Feature/ChatbotControllerTest.php [hunk 6]
```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace Tests\Feature;
+
+use Mockery;
```
Comment (ghost) [inline tests/Feature/ChatbotControllerTest.php]: cái này a ko thấy dùng ở phía dưới, e xem lại rồi remove đi nhé

File: tests/Unit/Repositories/BotRepositoryTest.php [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+     */
+    public function testGetAllByUser()
+    {
+        $botRepository = new BotRepository;
+        $botLists = factory(Bot::class, 10)->create(['name' => 'test get all bot by user']);
+        $user = $botLists[0]->user;
+        Auth::shouldReceive('user')->once()->andReturn($user);
+
+        $botRepository->getAllByUser();
+        $this->assertDatabaseHas('bots', ['name' => 'test get all bot by user']);
```
Comment (ghost) [inline tests/Unit/Repositories/BotRepositoryTest.php:42]: test_case này mình phải test data trả về của `$botRepository->getAllByUser()` có đúng với như mình mong muốn ko thì đúng hơn chứ nhỉ?
còn ở đây e test trong db có gì rồi

File: tests/Feature/ChatbotControllerTest.php (hunk 6 above)
Comment (nguyenthinhhn) [reply tests/Feature/ChatbotControllerTest.php]: ok a

File: tests/Unit/Repositories/BotRepositoryTest.php (hunk 7 above)
Comment (nguyenthinhhn) [reply tests/Unit/Repositories/BotRepositoryTest.php:42]: $botLists = factory(Bot::class, 10)->create(['name' => 'test get all bot by user']); 
chỗ này em tạo dữ liệu  insert vào
$botRepository->getAllByUser();
        $this->assertDatabaseHas('bots', ['name' => 'test get all bot by user']);
chỗ này em test hàm getall xem lấy ra xem có dữ liệu đó k

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr></table>

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr></table>

**Review by ghost — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Comment by phucqd-1305** [comment]
OK

**Pull request merged**